### PR TITLE
Dynamic BOM cost & structure report 

### PR DESCRIPTION
**Pull request by dut-odoo**
Description of the issue/feature this PR addresses:
Task: https://www.odoo.com/web?#id=36741&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.9bd934b064ebe9e34376ee506ac7b4fd

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
Dear @dut-odoo 

I seem that your Bom structure doesn't work if the template have some variante. In fact the Bom and Structure show all lines and gives the price of all line. When you show Bom and Structure you need to choose which `product.product`.
https://github.com/odoo/odoo/blob/master/addons/mrp/models/mrp_bom.py#L268

Exemple :
If I choose `Computer desk` with `Battery : Plomb`, I will see only `Table Legs` and `Bolt`

image

image

File: addons/mrp/models/mrp_bom_report.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                lines.update({
+                    'bom': bom,
+                    'bom_prod_name': bom.product_id.display_name or bom.product_tmpl_id.display_name or '',
+                    'currency': self.env.user.company_id.currency_id,
+                    'total': 0.0,
+                    'attribute': product.attribute_value_ids,
+                })
+                bom_lines = bom.bom_line_ids
+                if product.attribute_value_ids:
+                    bom_lines = bom.bom_line_ids.filtered(lambda l: product.attribute_value_ids in l.attribute_value_ids)
```
Comment (fmdl) [inline addons/mrp/models/mrp_bom_report.py]: Dear 

I will better to use this methode `_skip_bom_line(self, product)` to show or not the line : https://github.com/odoo/odoo/blob/10.0/addons/mrp/models/mrp_bom.py#L268

If this methode `_skip_bom_line(self, product)` is overpassed to explose Bom in Product Order (case in my company), the cost structure will be the same.

**Comment by fmdl** [comment]
@hbh-odoo thank for your IMP about the product variantes.

**Comment by hmo-odoo** [comment]
work continue on another branch: https://github.com/odoo-dev/odoo/tree/master-mrp-dynamic-bom-jat

**Pull request closed without merging**